frontend/src/metabase/status/components/FileUploadStatus/FileUploadStatus.tsx

```diff
@@ -1,39 +1,33 @@
+import _ from "underscore";
+
 import { useSelector, useDispatch } from "metabase/lib/redux";
 import { getAllUploads, clearAllUploads } from "metabase/redux/uploads";
-import Collections from "metabase/entities/collections/collections";
-import type { Collection } from "metabase-types/api";
+import type { CollectionId } from "metabase-types/api";
 import type { FileUpload } from "metabase-types/store/upload";
 import { isUploadAborted, isUploadInProgress } from "metabase/lib/uploads";
 
+import { useCollectionQuery } from "metabase/common/hooks";
 import useStatusVisibility from "../../hooks/use-status-visibility";
 import FileUploadStatusLarge from "../FileUploadStatusLarge";
 
-const FileUploadStatus = ({
-  collections = [],
-}: {
-  collections: Collection[];
-}) => {
+export const FileUploadStatus = () => {
   const uploads = useSelector(getAllUploads);
   const dispatch = useDispatch();
   const resetUploads = () => dispatch(clearAllUploads());
 
-  const uploadCollections = collections.filter(collection =>
-    uploads.some(upload => upload.collectionId === collection.id),
-  );
+  const groupedCollections = _.groupBy(uploads, "collectionId");
+
+  const collections = Object.keys(groupedCollections) as CollectionId[];
```

Comment:
unfortunately, underscore assumes that the keys pulled from `groupBy` will always be strings. This little bit of casting was the nicest way to get everything to play nicely.